client-cli/src/main/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationServiceCli.java

```diff
@@ -63,7 +78,21 @@ private void init() {
         } catch (IOException e) {
             throw new RuntimeException("Error reading file " + privateKeyFile, e);
         }
-        var apiClient = ClientUtils.createApiClient(service, clientDid, privateKeyData);
-        this.registryApiClient = new RegistryApi(apiClient);
+
+        // temporary to preserve the backwards compatibility
+        if (dataspaceDid.isEmpty()) {
+            var apiClient = ClientUtils.createApiClient(service, clientDid, privateKeyData);
+            this.registryApiClient = new RegistryApi(apiClient);
+            return;
+        }
+        var didWebResolver = new WebDidResolver(new OkHttpClient(), useHttpsScheme(), MAPPER, new ConsoleMonitor());
+        var urlResolver = new RegistrationUrlResolver(didWebResolver);
+        String registrationUrl = urlResolver.resolveUrl(dataspaceDid).orElseThrow(() -> new RuntimeException("Error resolving the registration url."));
+
```

Comment:
nit: just a suggestion - to improve reading if you agree we can move the registration service url resolving to its own method and use it within init() method.

Reply:
agree, done